src/eddymotion/data/filtering.py

```diff
@@ -0,0 +1,94 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2022 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Filtering data."""
+
+from __future__ import annotations
+
+import numpy as np
+from scipy.ndimage import median_filter
+from skimage.morphology import ball
+
+DEFAULT_DTYPE = "int16"
+"""The default image's data type."""
+
+
+def advanced_clip(
+    data: np.ndarray,
+    p_min: float = 35,
+    p_max: float = 99.98,
+    nonnegative: bool = True,
+    dtype: str | np.dtype = DEFAULT_DTYPE,
+    invert: bool = False,
+) -> np.ndarray:
+    """
+    Clips outliers from a n-dimensional array and scales/casts to a specified data type.
+
+    This function removes outliers from both ends of the intensity distribution
+    in a n-dimensional array using percentiles. It optionally enforces non-negative
+    values and scales the data to fit within a specified data type (e.g., uint8
+    for image registration). To remove outliers more robustly, the function
+    first applies a median filter to the data before calculating clipping thresholds.
+
+    Parameters
+    ----------
+    data : :obj:`~numpy.ndarray`
+        The input n-dimensional data array.
+    p_min : :obj:`float`, optional (default=35)
```

Comment:
I would not put the defaults in the docstrings: if for whatever reason they are changed in the method signature, there is a risk of not updating the docstrings. I've seen this happen in other projects, frequently.

Reply:
Yes, sounds reasonable. Please patch away :)